packages/opentelemetry-sampler-aws-xray/src/utils.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Attributes } from '@opentelemetry/api';
+
+// Template function from: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions#escaping
+const escapeRegExp = (regExPattern: string): string => {
+  // removed * and ? so they don't get escaped to maintain them as a wildcard match
+  return regExPattern.replace(/[.+^${}()|[\]\\]/g, '\\$&');
+};
+
+const convertPatternToRegex = (pattern: string): string => {
+  let regexPattern = '';
+
+  regexPattern = escapeRegExp(pattern).replace(/\*/g, '.*');
+  regexPattern = regexPattern.replace(/\?/g, '.');
+
+  return regexPattern;
+};
+
+export const wildcardMatch = (pattern?: string, text?: any): boolean => {
```

Comment:
```suggestion
export const wildcardMatch = (pattern?: string, text?: string): boolean => {
```

Since you use it as string, it might as well be typed correctly :)